crates/cxx-qt-lib/src/core/quuid.rs

```diff
@@ -0,0 +1,386 @@
+use crate::QString;
+use cxx::{type_id, ExternType};
+use std::fmt;
+#[cfg(feature = "uuid")]
+use uuid::Uuid;
+
+#[cxx::bridge]
+mod ffi {
+    #[repr(i32)]
+    #[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    enum QUuidVariant {
+        /// Variant is unknown
+        VarUnknown = -1,
+        /// Reserved for NCS (Network Computing System) backward compatibility
+        NCS = 0,
+        /// Distributed Computing Environment, the scheme used by QUuid
+        DCE = 2,
+        /// Reserved for Microsoft backward compatibility (GUID)
+        Microsoft = 6,
+        /// Reserved for future definition
+        Reserved = 7,
+    }
+
+    #[repr(i32)]
+    #[derive(Copy, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    enum QUuidVersion {
+        /// Version is unknown
+        VerUnknown = -1,
+        /// Time-based, by using timestamp, clock sequence, and MAC network card address (if
+        /// available) for the node sections
+        Time = 1,
+        /// DCE Security version, with embedded POSIX UUIDs
+        EmbeddedPOSIX = 2,
+        /// Name-based, by using values from a name for all sections
+        Md5 = 3,
+        /// Random-based, by using random numbers for all sections
+        Random = 4,
+        Sha1 = 5,
```

Comment:
SHA1 not good enough to get it's own documentation? :D

Reply:
Taken directory from the [Qt QUUID docs](https://doc.qt.io/qt-6/quuid.html#Version-enum)

Reply:
ah, that's fine then IMO since they omit it